spec/buildkite_upload_pipeline_action_spec.rb

```diff
@@ -0,0 +1,117 @@
+require 'spec_helper'
+
+describe Fastlane::Actions::BuildkiteUploadPipelineAction do
+  let(:pipeline_file) { 'path/to/pipeline.yml' }
+  let(:branch) { 'feature-branch' }
+  let(:commit) { 'abc123' }
+
+  before do
+    allow(File).to receive(:exist?).with(anything)
+  end
+
+  describe 'parameter validation' do
+    it 'raises an error when pipeline_file is not provided' do
+      expect do
+        run_described_fastlane_action(branch: branch)
+      end.to raise_error(FastlaneCore::Interface::FastlaneError, /pipeline_file/)
+    end
+
+    it 'raises an error when branch is not provided' do
+      expect do
+        run_described_fastlane_action(pipeline_file: pipeline_file)
+      end.to raise_error(FastlaneCore::Interface::FastlaneError, /branch/)
+    end
```

Comment:
As mentioned above, not sure this should be a requirement 🤔

Reply:
Updated on 7349eb36526b2d7e1d0b95beb822d0399ab4a6f4